### ISSUE TITLE
Implement rules for removing a consume/produce format that was previously supported

### ISSUE THREAD

**Issue by yyang08** [Yelp/swagger-spec-compatibility#20]
If the service no longer support a mime-type/data format, e.g. "application/json" or xml, it would be a breaking change for the users. There should be rules detecting these changes.

**Comment by macisamuele** [comment]
@yyang08 you're definitely correct

The tool was initially drafted to account for the "usual" backward incompatible changes that we were experiencing at Yelp 😅 .
Swagger specs are very expressive and flexible, but at the same time a lot of possible changes might break compatibility.
The idea of the tool is to provide a base-ground to implement the different detection rules.

I'm not sure on when I might have bandwidth to look into this by providing a PR, but we're always looking for and welcoming *contributors* 😄 

A good starting point for the implementation of new rules might be found in the [README](https://github.com/Yelp/swagger-spec-compatibility#how-to-define-a-new-compatibility-rule)

**Comment by yyang08** [comment]
@macisamuele Yes, this along with #21 are future improvements I want to get to at some point. Seems like it would require implementing new walkers for parsing the yaml spec. 

I agree that swagger-spec is a good tool for catching the most common types of breaking changes. My company is also trying to leverage this for better api development/versioning support. It's great that Yelp is open sourcing this. I really appreciate your work!
